core/field_textinput.js

```diff
@@ -145,20 +148,30 @@ Blockly.FieldTextInput.prototype.showEditor_ = function(opt_quietInput) {
 };
 
 /**
- * Handle a change to the editor.
+ * Handle key down to the editor.
  * @param {!Event} e Keyboard event.
  * @private
  */
-Blockly.FieldTextInput.prototype.onHtmlInputChange_ = function(e) {
+Blockly.FieldTextInput.prototype.onHtmlInputKeyDown_ = function(e) {
   var htmlInput = Blockly.FieldTextInput.htmlInput_;
-  if (e.keyCode == 13) {
-    // Enter
+  var enterKey = 13, escKey = 27;
+  if (e.keyCode == enterKey) {
     Blockly.WidgetDiv.hide();
-  } else if (e.keyCode == 27) {
-    // Esc
+  } else if (e.keyCode == escKey) {
     this.setText(htmlInput.defaultValue);
     Blockly.WidgetDiv.hide();
-  } else {
+  }
+};
+
+/**
+ * Handle a change to the editor.
+ * @param {!Event} e Keyboard event.
+ * @private
+ */
+Blockly.FieldTextInput.prototype.onHtmlInputChange_ = function(e) {
+  var htmlInput = Blockly.FieldTextInput.htmlInput_;
+  var escKey = 27;
+  if (e.keyCode != escKey) {
```

Comment:
I wrote `if (e.keyCode != enterKey && e.keyCode != escKey) {` then I got an issue on Firefox that didn't apply inputed text when I pressed enter key :sweat_smile: 
So, now I don't check about key code is enter key or not.